xbmc/music/tags/TagLoaderTagLib.cpp

```diff
@@ -462,6 +464,36 @@ bool CTagLoaderTagLib::ParseTag(ID3v2::Tag *id3v2, EmbeddedArt *art, MUSIC_INFO:
           tag.SetUserrating(POPMtoXBMC(popFrame->rating()));
         }
       }
+    else if (it->first == "CHAP")
+    {
+      //We use a chapter number index here to ensure chapters are added in the right order to the map
+      int chapNumber = 0;
+      int prevTime = 0;
+      for (ID3v2::FrameList::ConstIterator ch = it->second.begin(); ch != it->second.end(); ++ch)
+      {
+        auto* chapFrame = dynamic_cast<ID3v2::ChapterFrame*>(*ch);
+        if (!chapFrame)
+          continue;
+        std::string chapData = chapFrame->toString().toCString(true);
+        std::vector<std::string> chapInfo = StringUtils::Split(chapData, ",");
+        for (size_t i = 0; i + 2 < chapInfo.size(); i += 3)
+        { // splits to chapter name, start time( the string) and start time in millisecs
+          std::vector<std::string> titles = StringUtils::Split(chapInfo[i], ":");
+          // splits to end time (the string) and end time in millisecs
+          std::vector<std::string> timing = StringUtils::Split(chapInfo[i + 1], ":");
+          // Make sure last chapter really ends at file end
+          if (stoi(timing[1]) == prevTime)
```

Comment:
`stoi` can raise exceptions, use `std::from_chars` also other places

Reply:
Yeah thanks.  I wrote this originally about a year ago, but your comment made me re-visit it.   There is a cleaner way, not involving strings at all.